pelicanconf.py

```diff
@@ -35,24 +38,176 @@
+
+        if replace_string:
+            data = data.replace(citation_template.format(citation_name), replace_string)
+    
+    return data
+
+def remove_citations(data):
+    """ Remove citations from strings """
+
+    # Get citations names to remove from string
+    citation_names = get_citations(data)
+
+    for citation_name in citation_names:
+        data = data.replace("(Citation: " + citation_name + ")","")
+
+    return data
+
+no_stix_link_replacement = True
+def filter_urls(data):
+    """Filters out URLs to return path and not domain"""
+
+    global no_stix_link_replacement
+    if not no_stix_link_replacement:
+        if "https://attack.mitre.org/groups/" in data:
+            data = data.replace(
+                "https://attack.mitre.org/groups/", "/groups/")
+        if "https://attack.mitre.org/software/" in data:
+            data = data.replace(
+                "https://attack.mitre.org/software/", "/software/")
+        if "https://attack.mitre.org/techniques/" in data:
+            data = data.replace(
+                "https://attack.mitre.org/techniques/", "/techniques/")
+        if "https://attack.mitre.org/technique/" in data:
+            data = data.replace(
+                "https://attack.mitre.org/technique/", "/techniques/")
+    return data
+
+open_stix_replacement_file = False
+def stixToHTML(data, citations, firstParagraphOnly):
+    """ Clean output of STIX content.
+        params:
+            data (required, string), the STIX description to format
+            citations (optional, object), if not None, add citation markers to the data.
+            firstParagraphOnly (optional, boolean), if true, only return the first paragraph of the data in question.
+    """
+
+    global open_stix_replacement_file
+    global no_stix_link_replacement
+
+    # Open stix replacement settings file only once to get value
+    if not open_stix_replacement_file:
+        stix_replacement_file = os.path.join("data", "stix_replacement.js")
+        with open(stix_replacement_file, "r", encoding='utf8') as js_f:
+            no_stix_link_replacement = json.load(js_f)["no_stix_replacement"]
+            open_stix_replacement_file = True
+
+    # Replace data from markdown format
+    data = markdown.markdown(data)
+
+    # Replace url links
+    data = filter_urls(data)
+
+    # Get first paragraph from data
+    if firstParagraphOnly:
+        data = data.split('</p>')[0] + '</p>'
+        if data.startswith("<p>") and data.endswith("</p>"):
+            data = data[3:-4]
+
+    if citations:
+        # Update citations
+        data = update_citations(data, citations)
+    else:
+        # Remove citations
+        data = remove_citations(data)
+
+    data = clean_stix_data(data)
+
+    return data
+
 JINJA_FILTERS = {
     'from_json':json.loads,
     'flatten_tree': flatten_tree,
-    'clean_path': clean_path
+    'clean_path': clean_path,
+    'stixToHTML': stixToHTML
```

Comment:
Would it be possible to move the Jinja Filter function code into their own file and then import them here? I feel like it might be good to separate the actual Jinja configuration from all of our custom filter code.